app/experimenter/nimbus-ui/src/models/Experiment.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  gql,
+} from '@apollo/client';
+
+export const CREATE_EXPERIMENT_MUTATION = gql`
+  mutation createExperiment($input: CreateExperimentInput!) {
+    createExperiment(input: $input) {
+      clientMutationId
+      message
+      status
+      nimbusExperiment {
+        name
+        slug
+        hypothesis
+        application
+      }
+    }
+  }
+`;
+
+export type CreateExperimentInput = {
+  name: string;
+  hypothesis: string;
+  application: string;
+};
+
+export type CreateExperimentResult = {
+  status: number;
+  error: string;
+  message: string | Record<string, string[]>;
+  clientMutationId: string | null;
+  nimbusExperiment: {
+    name: string;
+    slug: string;
+    hypothesis: string;
+    application: string;
+  };
+};
```

Comment:
Yeah, I couldn't get `yarn types` to work... looking at it again, I think it's because I'm using a semi-dockerized setup, so my Django is running at `http://localhost:7000` and not the `--endpoint https://localhost` in package.json.

Are those types going to get checked in? If so, we might want to work some of that into CI. That is, fail if types haven't been generated for a commit. We have something like that for API docs where the job fails if checked-in docs differ from what a new generation run would produce

For `models` vs `gql` folder names, I was trying to follow what looks like the convention [from fxa-settings](https://github.com/mozilla/fxa/tree/main/packages/fxa-settings/src/models). But I can change that

Reply:
> I think it's because I'm using a semi-dockerized setup

Hmm, I'd love to use staging or prod so it doesn't require your local server to be up, but it is tucked under LDAP.

> Are those types going to get checked in?

I think they should be, unless we want to shove the command in front of the `yarn start` command so they are reliably generated each time.

Reply:
Okay, looks like I got the type generation to work locally. Seeing some wrinkles in the process, though.
```
npx apollo codegen:generate --target typescript --endpoint http://localhost:7001/api/v5/graphql types
```
There seem to be two types directories produced, one of which is outside of the `src/` directory:
```
types/globalTypes.ts
src/models/types/createExperiment.ts
```
Appears to work, but I seem to remember that CRA balked at code outside of `src/`.

Also wrestling a little bit with the generated types: `createExperiment_createExperiment_nimbusExperiment` has `slug` as a required field, but as far as I can tell it's not required on the Django side. 

Also the types all have `__typename` as a required constant, which doesn't seem correct? Not sure because I'm kind of still wrapping my head around this stuff.

And last thing I'm seeing now: the types are un-tested, so we probably want to exclude those files from code coverage (they come up as 0%).

Reply:
> For models vs gql folder names, I was trying to follow what looks like the convention from fxa-settings. But I can change that

Yeah totally, I guess I figured since it _only_ contained GQL stuff, and other utilities for the same record would be spread in to, for example, `hooks/`, it didn't really feel model-worthy. Definitely not a hill I'd die on though :)

Reply:
> I seem to remember that CRA balked at code outside of src/.

Hm, I know the error you're talking about, and I think that might be specific to component imports? Maybe not. However, I'm not seeing this error when I yarn build in my branch.

> Also the types all have __typename as a required constant

If the addition of __typename is causing issues we can def remove them with `--no-addTypename` ([docs](https://www.npmjs.com/package/apollo#apollo-clientcodegen-output))

Reply:
Moved the GQL to `gql/` and rolled in the generated types. Discovered that `createExperiment_createExperiment_nimbusExperiment` was the wrong type, using `CreateExperimentInput` from global types worked here.